service/src/test/java/bio/terra/drshub/controllers/DrsHubApiControllerTest.java

```diff
@@ -144,6 +146,48 @@ void testCallsCorrectEndpointsWhenOnlyAccessUrlRequestedWithPassportsUsingFallba
 
     // need an extra verify because nothing in the mock cares that bearer token is set or not
     verify(drsApi).setBearerToken(TEST_BOND_SA_TOKEN);
+    // verify that the passport postAccessURL method was not called, since there is no passport
+    verify(drsApi, never()).postAccessURL(any(), any(), any());
+  }
+
+  @Test
+  void testFallbackWhenOnlyAccessUrlRequestedWithPassportsFails() throws Exception {
+    var accessId = "gs";
+    var cidProviderHost = getProviderHosts("passport");
+    var drsObject = drsObjectWithRandomId("gs");
+
+    var drsApi =
+        mockDrsApiAccessUrlWithToken(
+            cidProviderHost.dnsHost(), drsObject, accessId, TEST_ACCESS_URL);
+
+    when(drsApi.postAccessURL(
+            Map.of("passports", List.of(TEST_PASSPORT)), drsObject.getId(), accessId))
+        .thenThrow(new RestClientException("Failed to retrieve access url with passport"));
+
+    mockExternalcredsApi("ras", TEST_ACCESS_TOKEN, Optional.of(TEST_PASSPORT));
+
+    mockBondLinkAccessTokenApi(
+        cidProviderHost.drsProvider().getBondProvider().get(),
+        TEST_ACCESS_TOKEN,
+        TEST_BOND_SA_TOKEN);
+
+    postDrsHubRequest(
+            TEST_ACCESS_TOKEN,
+            cidProviderHost.compactUriPrefix(),
+            drsObject.getId(),
+            List.of(Fields.ACCESS_URL))
+        .andExpect(status().isOk())
+        .andExpect(
+            content()
+                .json(
+                    objectMapper.writeValueAsString(Map.of(Fields.ACCESS_URL, TEST_ACCESS_URL)),
+                    true));
```

Comment:
I think tests should avoid duplication like other code we need to maintain. In this case this code fragment is repeated three other times in this file, and can be refactored into a common method.